http-server/src/main/java/io/micronaut/http/server/RouteExecutor.java

```diff
@@ -343,13 +340,13 @@ public MediaType resolveDefaultResponseContentType(HttpRequest<?> request, Route
                     return upstreamResponsePublisher;
                 }
                 HttpFilter httpFilter = filters.get(pos);
-                return Flux.from((Publisher<MutableHttpResponse<?>>) httpFilter.doFilter(requestReference.getAndSet(request), this))
+                return Flux.defer(() -> (Publisher<MutableHttpResponse<?>>) httpFilter.doFilter(requestReference.getAndSet(request), this))
```

Comment:
Using `Flux.defer` here somehow changing the thread pool which is getting used while subscription of elements from `default-nioEventLoopGroup` to `io-executor-thread-1`, I looked into java doc of `defer` it does not mentions anything about change in execution threads.

Another, but ugly way to fix this is, please do let me know if there is a better way to handle this.
```kotlin
 Flux<MutableHttpResponse<?>> responseFromFilter;
                try {
                    responseFromFilter = Flux.from((Publisher<MutableHttpResponse<?>>) httpFilter.doFilter(requestReference.getAndSet(request), this));
                } catch (Throwable t) {
                    responseFromFilter = Flux.error(t);
                }
                return responseFromFilter
                        .flatMap(handleStatusException)
                        .onErrorResume(onError);

Reply:
Catching the exception seems reasonable to me. You would just call `onError` though instead of applying the flatMap and onErrorResume

Reply:
The `flatMap` handling was present before my change, I think it is required, as it is handling the response of status code above `400`, while the `onErrorResponse` is handling other errors.

I added the commit to handle error and now test are passing.
Please trigger the github action.

Reply:
Handling a status code requires a response which we don't  have, so the onError is the only thing that needs to be called if the filter throws an exception. You can just return `onError.call(t)`

Reply:
In `flatMap` we do get `response`.

Reply:
flatMap would never be called because it processes emitted items and in this case there is no item being emitted

Reply:
Okay so you mean if status `400` and above the publisher emit error signal instead of response as next element signal.

Reply:
No - I mean `Flux.error(t).flatMap(() -> this never gets called)`

Reply:
Oh understood, implemented the suggested feedback.